docs/walkthroughs/local-setup.adoc

```diff
@@ -0,0 +1,179 @@
+* https://nodejs.org/en/[Node.js] >= 6.10.0
+
+*Note:* On Linux you also need to have the `libselinux-python` package
+installed!
+
+For local development, you will need to have a Go environment setup:
+
+[source,sh]
+----
+export PATH="$PATH:~/go/bin"
+----
+
+You will want to add the path permanently to your `.bashrc` or
+`.bashprofile`.
+
+[[creating-a-local-cluster]]
+Creating a Local Cluster
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+[[clone-this-repository]]
+Clone this repository
+^^^^^^^^^^^^^^^^^^^^^
+
+[source,bash]
+----
+git clone git@github.com:feedhenry/mcp-standalone.git
+----
+
+*Note:* it is a good idea to clone this into a valid $GOPATH, however it
+is not essential.
+
+[[install-the-required-ansible-dependencies]]
+Install the required ansible dependencies:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+[source,sh]
+----
+ansible-galaxy install -r ./installer/requirements.yml
+----
+
+[[firewall-setup]]
+Firewall setup
+^^^^^^^^^^^^^^
+
+Next we need to configure Docker registry _and_ ports required as part
+of the cluster setup: * Linux: Follow steps 2 _and_ 3
+https://github.com/openshift/origin/blob/master/docs/cluster_up_down.md#linux[here]
+* Mac: Follow steps 2 _and_ 3
+https://github.com/openshift/origin/blob/master/docs/cluster_up_down.md#macos-with-docker-for-mac[here]
+
+For Linux (Fedora) we also need to add an extra port to the `dockerc`
+zone:
+
+[source,sh]
+----
+firewall-cmd --permanent --zone dockerc --add-port 443/tcp
+firewall-cmd --reload
+----
+
+[[run-the-ansible-installer]]
+Run the ansible installer
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This installer sets up your OpenShift environment with the service catalog and the ansible service broker.
+
+[source,sh]
+----
+export DOCKERHUB_USERNAME="<username>"
+export DOCKERHUB_PASSWORD="<password>"
+cd installer/ && ansible-playbook playbook.yml -e "dockerhub_username=$DOCKERHUB_USERNAME" -e "dockerhub_password=$DOCKERHUB_PASSWORD" --ask-become-pass
+----
+
+We ask for a DockerHub username and password currently because the
+Ansible Service Broker calls Docker APIs to read/load Ansible Playbook
+Bundles from a Docker Hub organisation (defaults to `feedhenry`).
+
+This will set up your cluster for you - note that it is possible for
+this to fail on the first attempt, as the cluster up check may fail
+waiting for the images to be pulled - if this happens, run
+`oc cluster down` and execute the playbook again.
+
+*Optional*
+
+You can enable Hawkular metrics by passing `-e 'hawkular_metrics=true'`
+as part of your `ansible-playbook` command.
+
+[[local-development-only-run-the-mcp-server-locally]]
+(Local Development Only) Run the MCP Server locally
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For local development, you can run the MCP server locally instead of in
+OpenShift. This allows for faster development feedback. To compile the
+MCP Server execute the following in the root directory of the repo:
+
+[source,sh]
+----
+cd ..
+make run_server NAMESPACE=localmcp
+----
+
```

Comment:
Should we briefly explain what this will result in? The localmcp namespace being created if it doesn't already exist etc.

Reply:
:+1: